firmware/application/freqman.cpp

```diff
@@ -103,51 +103,48 @@ std::vector<std::string> get_freqman_files() {
 };
 
 bool load_freqman_file(std::string& file_stem, freqman_db& db) {
-    return load_freqman_file_ex(file_stem, db, true, true, true);
+    return load_freqman_file_ex(file_stem, db, true, true, true, FREQMAN_MAX_PER_FILE);
 }
 
-bool load_freqman_file_ex(std::string& file_stem, freqman_db& db, bool load_freqs, bool load_ranges, bool load_hamradios) {
+bool load_freqman_file_ex(std::string& file_stem, freqman_db& db, bool load_freqs, bool load_ranges, bool load_hamradios, uint8_t limit) {
+    // these are not enough to really start with a new, clean, empty vector
+    // swap is the only way to achieve a perfect memory liberation
+    std::vector<freqman_entry>().swap(db);
+
     File freqman_file;
     size_t length, n = 0, file_position = 0;
     char* pos;
     char* line_start;
     char* line_end;
     std::string description;
     rf::Frequency frequency_a, frequency_b;
-    const int read_buf_size = 128;
-    char file_data[read_buf_size + 1];
+    char file_data[FREQMAN_READ_BUF_SIZE + 1];
     freqman_entry_type type;
     freqman_index_t modulation = 0;
     freqman_index_t bandwidth = 0;
     freqman_index_t step = 0;
     freqman_index_t tone = 0;
 
-    // these are not enough to really start with a new, clean, empty vector
-    // swap is the only way to achieve a perfect memory liberation
-    // db.clear();
-    // db.shrink_to_fit():
-    std::vector<freqman_entry>().swap(db);
-
     auto result = freqman_file.open("FREQMAN/" + file_stem + ".TXT");
     if (result.is_valid())
         return false;
 
     while (1) {
-        // Read a read_buf_size bytes block from file
+        // Read a FREQMAN_READ_BUF_SIZE block from file
         freqman_file.seek(file_position);
 
-        memset(file_data, 0, read_buf_size + 1);
-        auto read_size = freqman_file.read(file_data, read_buf_size);
+        memset(file_data, 0, FREQMAN_READ_BUF_SIZE + 1);
```

Comment:
nit: could also just null terminate at `*read_size` if you didn't want to zero the buffer each pass.

Reply:
I could, but I like my memory buffers to be clean ;-)